src/cpp/server/DBActiveSessionStorage.cpp

```diff
@@ -0,0 +1,404 @@
+         if(++iter != rowset.end())
+         {
+            return Error{"Too many sessions returned", errc::TooManySessionsReturned, ERROR_LOCATION};
+         }
+
+         database::Query updateQuery = connection->query("UPDATE "+kTableName+" SET "+getUpdateString(properties)+" WHERE session_id = :id")
+            .withInput(sessionId_);
+         
+         error = connection->execute(updateQuery);
+         if(error)
+         {
+            return Error{"DatabaseException", errc::DBError, "Error while updating properties [ session:"+sessionId_+" properties:"+getKeyString(properties)+" ]", error, ERROR_LOCATION};
+         }
+      }
+      else
+      {
+         database::Query insertQuery = connection->query("INSERT INTO "+kTableName+" ("+kSessionIdColumnName+", "+getKeyString(properties)+") VALUES (:id, "+getValueString(properties)+")")
+            .withInput(sessionId_);
+
+         error = connection->execute(insertQuery);
+         if(error)
+         {
+            return Error{"DatabaseException", errc::DBError, "Error while updating properties [ session:"+sessionId_+" properties:"+getKeyString(properties)+" ]", error, ERROR_LOCATION};
+         }
+      }
+   }
+   return error;
+}
+
+Error DBActiveSessionStorage::destroy()
+{
+   boost::shared_ptr<database::IConnection> connection;
+   Error error = getConnectionOrOverride(&connection);
+
+   if(error)
+   {
+      return error;
+   }
+
+   database::Query query = connection->query("DELETE FROM "+kTableName+" WHERE "+kSessionIdColumnName+" = :id")
+      .withInput(sessionId_);
+   database::Rowset rowset{};
+
+   error = connection->execute(query, rowset);
+
+   if(!error)
+   {
+      return Error{"DatabaseException", errc::DBError, "Error while deleting session metadata [ session:"+sessionId_+" ]", error, ERROR_LOCATION};
+   }
+   else
+   {
+      return error;
+   }
+}
+
+
+Error DBActiveSessionStorage::isValid(bool* pValue)
+{
+   boost::shared_ptr<database::IConnection> connection;
+   Error error = getConnectionOrOverride(&connection);
+
+   if(error)
+   {
+      return error;
+   }
+
+   database::Query query = connection->query("SELECT COUNT(*) FROM "+kTableName+" WHERE "+kSessionIdColumnName+" = :id")
+      .withInput(sessionId_);
+   database::Rowset rowset{};
+
+   *pValue = false;
+
+   error = connection->execute(query, rowset);
+
+   if(error)
+   {
+      return Error{"DatabaseException", errc::DBError, "Error while deleting session metadata [ session:"+sessionId_+" ]", error, ERROR_LOCATION};
+   }
+   database::RowsetIterator iter = rowset.begin();
+
+   // Sanity checking, but should always have 1 row containing the count of rows
+   if(iter != rowset.end())
+   {
+      int count = iter->get<int>("COUNT(*)");
+
+      // ensure one and only one
+      if(count > 1)
+      {
+         return Error{"Too Many Sessions Returned", errc::TooManySessionsReturned, ERROR_LOCATION};
```

Comment:
I suspect this message won't give enough information. Could you add the amount we found and at least the ID of the session that we were looking for?